plugins/discovery/discovery.go

```diff
@@ -35,17 +36,18 @@ const Name = "discovery"
 // started it will periodically download a configuration bundle and try to
 // reconfigure the OPA.
 type Discovery struct {
-	manager      *plugins.Manager
-	config       *Config
-	factories    map[string]plugins.Factory
-	downloader   *download.Downloader                // discovery bundle downloader
-	status       *bundle.Status                      // discovery status
-	listenersMtx sync.Mutex                          // lock for listener map
-	listeners    map[interface{}]func(bundle.Status) // listeners for discovery update events
-	etag         string                              // discovery bundle etag for caching purposes
-	metrics      metrics.Metrics
-	readyOnce    sync.Once
-	logger       logging.Logger
+	manager            *plugins.Manager
+	config             *Config
+	factories          map[string]plugins.Factory
+	downloader         *download.Downloader                // discovery bundle downloader
+	status             *bundle.Status                      // discovery status
+	listenersMtx       sync.Mutex                          // lock for listener map
+	listeners          map[interface{}]func(bundle.Status) // listeners for discovery update events
+	etag               string                              // discovery bundle etag for caching purposes
+	metrics            metrics.Metrics
+	readyOnce          sync.Once
+	logger             logging.Logger
+	prometheusRegister prometheus.Registerer
```

Comment:
Sorry for the back and forth, but in my head, it would perhaps look kind of nice if the prometheus.Registerer mechanism was part of the plugin manager: every plugin could then register its metrics with the plugin manager. What do you think?

Reply:
It looks nice, I've seen an issue about that https://github.com/open-policy-agent/opa/issues/2348. I will try to implement the interface prometheus.Registerer on the manager.